src/raw/dockerapi/fetcher.go

```diff
@@ -35,24 +37,43 @@ func (f *Fetcher) Fetch(container types.ContainerJSON) (raw.Metrics, error) {
 	return metrics, nil
 }
 
-func (f *Fetcher) memoryMetrics(containerStats types.StatsJSON) raw.Memory {
-	panic("unimplemented")
+func (f *Fetcher) memoryMetrics(containerStats types.StatsJSON, hostConfig *container.HostConfig) raw.Memory {
+	mem := raw.Memory{}
+
+	if containerStats.MemoryStats.Usage != 0 {
+		mem.UsageLimit = containerStats.MemoryStats.Limit
+		mem.FuzzUsage = containerStats.MemoryStats.Usage
+	}
+
+	if hostConfig != nil {
```

Comment:
Could `hostConfig` be nil? If it could, should we log a debug message?

Reply:
Yes, it can be nil https://github.com/moby/moby/blob/master/api/types/types.go#L315C2-L315C39. Make sense, going to add a debug log in case of nil.